components/style/servo/url.rs

```diff
@@ -126,3 +127,33 @@ impl ToCss for SpecifiedUrl {
         dest.write_str(")")
     }
 }
+
+impl ToComputedValue for SpecifiedUrl {
+    type ComputedValue = ComputedUrl;
+
+    // If we can't resolve the URL from the specified one, we fall back to the original
+    // but still return it as a ComputedUrl::Invalid
+    fn to_computed_value(&self, _: &Context) -> Self::ComputedValue {
+        match self.resolved {
+            Some(ref url) => ComputedUrl::Valid(url.clone()),
+            None => match self.original {
+                Some(ref url) => ComputedUrl::Invalid(url.clone()),
+                None => ComputedUrl::Invalid(Arc::new("about:invalid".to_string())),
```

Comment:
I think this is the same case as the `ToCss` implementation for `SpecifiedUrl`, when there's no original and no resolved:
```
// This can only happen if the url wasn't specified by the
// user *and* it's an invalid url that has been
// back to specified value via the "uncompute" functionality.
```
So I think the case is reachable, but I'm not sure the behavior is appropriate.